controllers/vspherevm_controller_test.go

```diff
@@ -230,13 +229,13 @@ func TestReconcileNormal_WaitingForIPAddrAllocation(t *testing.T) {
 			Network:  nil,
 		}, nil)
 		r := setupReconciler(fakeVMSvc)
-		_, err = r.Reconcile(context.Background(), ctrl.Request{NamespacedName: util.ObjectKey(vsphereVM)})
+		_, err = r.Reconcile(ctx, ctrl.Request{NamespacedName: util.ObjectKey(vsphereVM)})
```

Comment:
```suggestion
		_, err = r.Reconcile(context.Background(), ctrl.Request{NamespacedName: util.ObjectKey(vsphereVM)})
```

Let's revert them here,

the others were indirectly already using the package-wide context, not here. Let's keep theme explicitly at `context.Background()` to not change behaviour.

Reply:
All reverted, will refine them in a separate task as we discussed.